lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2110,7 +2110,9 @@ Instruction *SPIRVToLLVM::transEnqueueKernelBI(SPIRVInstruction *BI,
   bool HasVaargs = Ops.size() > 10;
   bool HasEvents = true;
   SPIRVValue *NumEvents = Ops[3];
-  if (NumEvents->getOpCode() == OpConstant) {
+  SPIRVValue *EventRet = Ops[5];
+  if (EventRet->getOpCode() == OpConstantNull &&
+      NumEvents->getOpCode() == OpConstant) {
```

Comment:
I guess `NumEvents` as well can be an `OpConstantNull`. Should we take it into account ?

Reply:
I don't know if NumEvents can be OpConstantNull. @AlexeySachkov @aratajew do you have any comments?

Reply:
Current support for enqueue_kernel translation from LLVM to SPIRV does not produce OpConstantNull. It produces OpConstant with 0 value. But OpConstantNull as an NumEvents argument is also correct in SPIRV, so we should take it into account, indeed.

Reply:
I've updated the code.